Sources/MessagingInApp/Gist/Gist.swift

```diff
@@ -126,6 +139,41 @@ public class Gist: GistDelegate {
             }
     }
 
+    // If someone sets a page rule on a message, they want the message to show on that screen. Because messages can take multiple seconds to finish rendering, there is a chance that
+    // a user navigates away fron a screen when the rendering finishes. To fix this, cancel showing a modal message if a message is still loading.
+    //
+    // Like dismiss message, but does not call event listener.
+    // Dismiss the currently shown message, if there is one, and then remove message manager allowing us to show a message again in the future.
+    func cancelLoadingModalMessage() {
+        guard let messageManagerToCancel = getModalMessageManager() else {
+            return // no message being shown or loading.
+        }
+        let currentMessage = messageManagerToCancel.currentMessage
+
+        if messageManagerToCancel.isShowingMessage {
+            // The modal is already visible, don't cancel it.
+            // This can prevent an infinite loop scenario:
+            // * page rule changed and that triggers showing a Modal
+            // * Modal message is displayed on screen
+            // * Modal being displayed triggers an auto screenview tracking event. This triggers a SDK page route change
+            // * Request to cancel modal message
+            // * Back to the foreground screen that originally triggered showing a Modal message...repeat...
+            return
+        }
+
+        guard currentMessage.gistProperties.routeRule != nil else {
```

Comment:
At first, I thought that a regex check was overkill because the logic in `setCurrentRoute` detects when the route *changes* which would mean that the user navigated to a different screen and as long as a message has a page rule attached, we could assume that the new screen would not match the modal's page rule. 

However, you helped me realize that you can use wildcards in page rules. So, there is indeed a chance that you change the screen and it does match the page rule of the message. 

I'll fix this in another PR.